### PR TITLE
Throw a readable error when the entry file does not exist

### PR DESCRIPTION
**Pull request by olane**
This addresses #27.

I've fixed a test that failed on Windows for the same reason I was running into problems. I've also thrown a descriptive error when the file does not exist, as suggested in the issue thread, and written a test for that.

Finally, I've updated the readme to reflect how to create a configuration that works across platforms with nested paths.

Happy to tweak the wording of the documentation/error message if you'd like.

### CODE REVIEW

**Comment by olane** [comment]
Seems to be failing on the build server but not locally - I thought it might be the rejected promise vs. throwing thing in 2fcde79 but it seems not.

**Comment by ghost** [comment]
I think it's because the `'file-does-not-exist-error'` folder doesn't exist in `test/fixtures`. At least not when I check out your PR.

**Comment by olane** [comment]
Sigh.. thanks.

File: lib/index.js [hunk 1]
```diff
@@ -12,6 +13,13 @@ const bundle = ({ file, options, source, files }) => {
 
   const readable = new Readable();
   const filePath = path.join(source, file);
+
+  if (!fs.existsSync(filePath)) {
+    return Promise.reject(
+      new Error(`Cannot use '${file}' as an entry point - file does not exist.`)
```
Comment (ghost) [inline lib/index.js]: Maybe `Can't find '${file}'. The file doesn't exist or the path is invalid.` would be a bit more descriptive here.

File: README.md [hunk 2]
```diff
@@ -46,6 +46,8 @@ The entry points that should be browserified. So this `metalsmith.json`:
 
 Would browserify both `./src/index.js` and `./src/another.js` and output them as `./build/index.js` and `./build/another.js` respectively.
 
+Note that if the entry path is nested, the paths may differ across environments (for example, on Windows the paths may use `\` as a separator). To fix this, use the Javascript API and use `path.join` to generate the correct path for your system at runtime.
```
Comment (ghost) [inline README.md]: Reworded it slightly:

> Note that if the entry path is nested, the paths may differ across operating systems. Make sure you're using the correct directory separators, or use node's [path.join](https://nodejs.org/api/path.html#path_path_join_paths) to make sure the path will work anywhere.

What do you think?

**Review by ghost — COMMENTED** [review]
Thanks for the pr! Looks good. I left comments on the readme and the error message. Let me know what you think.

**Comment by ghost** [comment]
Took the liberty to apply the suggested changes, rebase and add the necessary code to implement what we discussed in the related issue. Thanks for suggesting this @olane! Hope this does what you were after, and let me know if you have any feedback!

**Comment by olane** [comment]
Awesome, thanks for sorting it out! Pretty sure it does what I needed it to, I'll file another issue if I run into any problems.

**Comment by ghost** [comment]
Cool, thanks!

**Pull request merged**